src/main/java/org/radarcns/mongo/util/MongoHelper.java

```diff
@@ -81,32 +77,30 @@ public String getParam() {
     }
 
     /**
-     * Finds all Documents within [start-end] belonging to the given subject for the give
-     * sourceType.
+     * Finds all documents within a time window belonging to the given subject, source and project.
      *
-     * @param subject is the subjectID
-     * @param source is the sourceID
+     * @param projectName of the project
+     * @param subjectId is the subjectID
+     * @param sourceId is the sourceID
      * @param start is the start time of the queried timewindow
      * @param end is the end time of the queried timewindow
      * @param collection is the MongoDB that will be queried
      * @return a MongoDB cursor containing all documents from the query.
      */
     public static MongoCursor<Document> findDocumentsByProjectAndSubjectAndSourceInWindow(
-            String projectName, String subject, String source, Long start, Long end,
+            String projectName, String subjectId, String sourceId, Date start, Date end,
             MongoCollection<Document> collection) {
-        FindIterable<Document> result = collection.find(
-                Filters.and(
-                        eq(PROJECT_ID, projectName),
-                        eq(USER_ID, subject),
-                        eq(SOURCE_ID, source),
-                        gte(START, new Date(start)),
-                        lte(END, new Date(end)))).sort(new BasicDBObject(START, 1));
+        BasicDBObject query = getByProjectSubjectSource(projectName, subjectId, sourceId)
+                .append(KEY.concat(".").concat(START), new BasicDBObject("$gte", start))
```

Comment:
unless strings are expected to be empty, prefer `KEY + "." + START` over two concats.

Reply:
Thanks for comments. Will look into those soon.